RecoLocalCalo/HGCalRecAlgos/interface/HGCalUncalibRecHitRecWeightsAlgo.h

```diff
@@ -68,17 +71,14 @@ class HGCalUncalibRecHitRecWeightsAlgo {
         // LG (11/04/2016):
         // offset the TDC upwards to reflect the bin center
         amplitude_ = (std::floor(tdcOnsetfC_ / adcLSB_) + 1.0) * adcLSB_ + (double(sample.data()) + 0.5) * tdcLSB_;
-
-        if (sample.getToAValid()) {
-          jitter_ = double(sample.toa()) * toaLSBToNS_;
-        }
       } else {
         amplitude_ = double(sample.data()) * adcLSB_;  // why do we not have +0.5 here ?
-        if (sample.getToAValid()) {
-          jitter_ = double(sample.toa()) * toaLSBToNS_;
-        }
-      }  //isSiFESim_
-    }    //mode()
+      }                                                //isSiFESim_
+      if (sample.getToAValid() && geom_) {
+        const auto& dist2center = geom_->getPosition(dataFrame.id()).mag();
+        jitter_ = double(sample.toa()) * toaLSBToNS_ - dist2center / c_cm_ns - tofDelay_;
```

Comment:
What is going to happen if the `getToAValid()` is true but the `geom_` pointer is null?

Reply:
If getToaValid is true but geom_ is null, then that means that the hit has an assigned time but no information is derived for the distance, as such, only the global time (i.e. at the detector) would be known for that hit. However, I am not sure in which circumstance this can happen. I added the geom_ check to ensure that the point is always non null when used.

Reply:
I understand the check on the validity of the pointer before using it. On the other hand, it would be nice to understand if really this condition could happen in "real life", why and how to handle that. 
Was that logic present in the previous implementation?
No time assigned at all could be maybe worse than a non-tof-correced time?

Reply:
Before we did not do time correction at the uncalibrated rec hits, so using the geometry in the uncalibrated rec hits was not crucial.
Currently the only case in which the geometry is null is when the HF nose is not set (some geometry profiles), in which case the time should also not exist (since it is not ran anyways).

I can update the PR only checking the geom_ when filling the distance to center:

const auto& dist2center = geom_ ? geom_->getPosition(dataFrame.id()).mag() : 0;